ui/components/ui/update-nickname-popover/update-nickname-popover.js

```diff
@@ -0,0 +1,109 @@
+import React, { useCallback, useContext } from 'react';
+import PropTypes from 'prop-types';
+
+import Popover from '../popover';
+import Button from '../button';
+import TextField from '../text-field';
+
+import { I18nContext } from '../../../contexts/i18n';
+
+import Identicon from '../identicon/identicon.component';
+
+export default function UpdateNicknamePopover({
+  nickname,
+  address,
+  onAdd,
+  memo,
+  onClose,
+}) {
+  const t = useContext(I18nContext);
+
+  const closePopover = useCallback(() => {
+    onClose();
+  }, [onClose]);
+
+  const onCancel = () => {
+    onClose();
+  };
+
+  const onSubmit = useCallback(() => {
+    onAdd(address);
```

Comment:
`memo` should be passed to `onAdd` as well

Reply:
Or rather, we need to make sure that `memo` and `nickname`, from the input fields, are passed to `onAdd` on submission

So we might need to maintain some component state here so that we can capture what the user enters and pass it along on submission

Reply:
Thought we would add this when component is added to the app. I've just added the changes.